src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -933,54 +914,31 @@ private Task WriteToStreamAsync(ReadOnlyMemory<byte> source, CancellationToken c
 
         private async ValueTask<ArraySegment<byte>> ReadNextLineAsync(CancellationToken cancellationToken)
         {
-            int searchOffset = 0;
+            int previouslyScannedBytes = 0;
             while (true)
             {
-                int startIndex = _readOffset + searchOffset;
-                int length = _readLength - startIndex;
-                int crPos = Array.IndexOf(_readBuffer, (byte)'\r', startIndex, length);
-                if (crPos < 0)
-                {
-                    // Couldn't find a \r.  Read more.
-                    searchOffset = length;
-                    await FillAsync(cancellationToken).ConfigureAwait(false);
-                }
-                else if (crPos + 1 >= _readLength)
+                int scanOffset = _readOffset + previouslyScannedBytes;
+                int lfIndex = Array.IndexOf(_readBuffer, (byte)'\n', scanOffset, _readLength - scanOffset);
+                if (lfIndex >= 0)
                 {
-                    // We found a \r, but we don't have enough data buffered to read the \n.
-                    searchOffset = length - 1;
-                    await FillAsync(cancellationToken).ConfigureAwait(false);
-                }
-                else if (_readBuffer[crPos + 1] == '\n')
-                {
-                    // We found a \r\n.  Return the data up to and including it.
-                    int lineLength = crPos - _readOffset + 2;
-                    var result = new ArraySegment<byte>(_readBuffer, _readOffset, lineLength);
-                    _readOffset += lineLength;
-                    return result;
-                }
-                else
-                {
-                    ThrowInvalidHttpResponse();
+                    int startIndex = _readOffset;
+                    int length = lfIndex - startIndex;
+                    if (length > 0 && _readBuffer[startIndex + length - 1] == '\r')
```

Comment:
I know you reworked this and we had a whole go around on it, so please feel free to do whatever you think is best here.  FWIW, I think this looks funny... `length` was just set to `lfIndex - startIndex`, so `startIndex + length - 1` is just `startIndex + lfIndex - startIndex - 1`, or just `lfIndex - 1`.  So I still don't understand why it can't just be `if (lfIndex != 0 && _readBuffer[lfIndex - 1] == '\r')`, but it's not worth any more discussion time :)